src/rules/no-restricted-paths.js

```diff
@@ -75,13 +77,35 @@ module.exports = {
         }
 
         matchingZones.forEach((zone) => {
-          const exceptionPaths = zone.except || []
+          const exceptions = zone.except || []
           const absoluteFrom = path.resolve(basePath, zone.from)
+          let exceptionPaths = []
+          let exceptionGlobs = []
+
+          exceptions.forEach(exception => {
+            if (isGlob(exception)) {
```

Comment:
is this not something minimatch can validate?

Reply:
I just looked into it, but unfortunately `minimatch` is not enough here. It would just tell me that it does not match, but it doesn't tell me if it is because the glob is not matching or because it cannot handle relative paths. If I don't know that I would also not know when to call `reportInvalidExceptionPath`.
So I guess we have to use isGlob here.

Reply:
what do you mean by "cannot handle relative paths"? is there a need for a different error message between globs and paths?

Reply:
Sorry, maybe I wasn't clear enough.
`isGlob` checks if it is a glob and thus a relative path like "./server" will be marked as non-glob ("exceptionPaths"). Otherwise if I would check for example "./server" with minimatch it will tell me this path doesn't match, no matter if this path would actually match, because minimatch only works with globs.

Also the check if you have an invalid exception path is done by `path.relative()`, which also does not work with globs. That's why you will currently not see the _"Restricted path exceptions must be descendants of the configured `from` path for that zone."_ error message for any glob you provide.
And I think it is totally ok, because you would use globs for different reasons anyways. For example if you want to exclude all "**/types.js", you cannot really check if this is a descendant (you could check if there is any file that matches, but this would take a lot of resources and wouldn't make sense to me).